CRM/Queue/Queue.php

```diff
@@ -60,7 +62,17 @@ public function __construct($queueSpec) {
    */
   public function isActive(): bool {
     $status = CRM_Core_DAO::getFieldValue('CRM_Queue_DAO_Queue', $this->_name, 'status', 'name', TRUE);
-    // Note: In the future, we may want to incorporate other data (like maintenance-mode or upgrade-status) in deciding active queues.
+    if ($status === 'active' && \Civi::settings()->get('queue_maintenance_mode')) {
+      $status = 'deferred';
+    }
+    $event = GenericHookEvent::create([
+      'status' => &$status,
+      'queue_name' => $this->_name,
+      'queue_spec' => $this->queueSpec,
+    ]);
+    \Civi::dispatcher()->dispatch('is_queue_active', $event);
```

Comment:
If you run `cv event /queue/`, you'll see some existing event-names look like:

```
[Event] civi.queue.check
[Event] civi.queue.runTask.start
[Event] hook_civicrm_queueRun_task
[Event] hook_civicrm_queueStatus
```

So maybe `civi.queue.isActive` or `hook_civicrm_queueActive`?

Reply:
@totten do you have a sense of which of the above it is closest to logically? My instinct is perhapes queueStatus

Reply:
@totten I've updated to the first suggestion in the spirit of 'pick one any one'

Reply:
Yeah, it just depends how public one wants to make it. The first one (`civi.queue.isActive`; less public) sounds good.

Weird thing - Github UI isn't showing the update. (Well, it shows a force-push, but the main diff hasn't changed.) Maybe if we "Resolve conservation" it will refresh?

Reply:
Nope that, didn't help. Weird.

Reply:
So I see 

![image](https://github.com/civicrm/civicrm-core/assets/336308/8182da12-edc3-40e3-a5e1-c24c667f1240)